### PR TITLE
CORE-4877 deleteById/findAll persistence methods

### PR DESCRIPTION
**Pull request by barrylapthorn**
See also: https://github.com/corda/corda-runtime-os/pull/1354

Moved Avro messages to own package.

Added DeleteById and FindAll persistence methods.

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteById.avsc [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "doc": "Delete an entity by its id.  See also {@link EntityRequest#request}",
+  "namespace": "net.corda.data.persistence",
+  "fields": [
+    {
+      "name": "entityClassName",
+      "type": "string",
+      "doc": "the fully qualified entity class name"
+    },
+    {
+      "name": "id",
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteById.avsc]: We use `primaryKey` as a name for the `FindEntity` schema. Maybe it's worth bringing them in line?

**Review by driessamyn — COMMENTED** [review]
Just a small comment about naming consistency.

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteEntityByPrimaryKey.avsc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "type": "record",
+  "name": "DeleteEntityByPrimaryKey",
+  "doc": "Delete an entity by its id.  See also {@link EntityRequest#request}",
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteEntityByPrimaryKey.avsc]: I think `DeleteEntityById` was actually correct. It's the JPA object `@Id` annotation that is relevant, it doesn't actually _have_ to be a primary key in the Db.

Comment (barrylapthorn) [reply data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteEntityByPrimaryKey.avsc]: So you want it changed back to DeleteById then?

Comment (driessamyn) [reply data/avro-schema/src/main/resources/avro/net/corda/data/persistence/DeleteEntityByPrimaryKey.avsc]: I think that would make more sense in terms of an API.
Sorry, I should have been clearer in my original comment.

**Review by driessamyn — DISMISSED** [review]
LGTM

**Review by driessamyn — APPROVED** [review]

**Pull request merged**